test/integrations/testUtils.ts

```diff
@@ -104,6 +97,42 @@ export const overrideDestination = (destination: Destination, overrideConfigValu
   });
 };
 
+export const produceTestData = (testData: TestCaseData[]) => {
+  const result: any = [];
+  testData.forEach((tcData) => {
+    let events;
+    try {
+      switch (tcData.feature) {
```

Comment:
Are you sure that the payload server sends to the transformer is fully compatible with what clients send to the data plane?

Also, should we exclude the negative test cases?

Reply:
I don't think we should exclude any specific test from the tool perspective we have already share flexibility to consider specific test with the usage of CLI options also negative tests could also be required for E2E testing.

As server compatibility, AFAIK only specific fields like `rudderId`, `timestamp` fields like `originalTimeStamp` if not sent from client server computes and adds them otherwise the payloads are compatible

Reply:
Resolve this comment once you verify the server behavior with payloads containing the reserved fields.

Reply:
Also added support to filter such keys